### PR TITLE
pinniped: init at 0.17.0

### PR DESCRIPTION
**Pull request by bpaulin**
###### Description of changes

Adding https://github.com/vmware-tanzu/pinniped "Pinniped is the easy, secure way to log in to your Kubernetes clusters."

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/pinniped/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-0h7zyKe2gmC1n9EB5FRVI/io7Yj+91ZAtLy+1u3gyO0=";
+  };
+
+  subPackages = "cmd/pinniped";
+
+  vendorSha256 = "sha256-8ohyyciL1ORYOxPu64W0jXASTv+vVZR8StutzbF9N4Y=";
+
+  meta = with lib; {
+    description = "Pinniped is the easy, secure way to log in to your Kubernetes clusters.";
```
Comment (fabaff) [inline pkgs/applications/networking/cluster/pinniped/default.nix:19]:
```suggestion
    description = "Tool to securely log in to your Kubernetes clusters";
```

see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

**Review by fabaff — COMMENTED** [review]
Rest looks good to me.

**Review by viraptor — COMMENTED** [review]
Result of `nixpkgs-review pr 176882` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pinniped</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/pinniped/default.nix (hunk 1 above)
Comment (bpaulin) [reply pkgs/applications/networking/cluster/pinniped/default.nix]: thanks, done

**Comment by onny** [comment]
Please rebase on top of master

**Comment by onny** [comment]
Result of `nixpkgs-review pr 176882` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pinniped</li>
  </ul>
</details>

**Pull request merged**